generator/src/test/java/com/scottlogic/deg/generator/cucumber/longerThan.feature

```diff
@@ -0,0 +1,671 @@
+  | "aa" |
+  | "aaa" |
+  | "aaaa" |
+  | "aaaaa" |
+  | "aaaaaa" |
+  Then the following data should be generated:
+    | foo  |
+    | null |
+    | "aaaaaa" |
+
+Scenario: Running a 'longerThan' request that includes the value zero should be succesful
+  And foo is longer than 0
+  And foo is in set:
+  | "" |
+  | "a" |
+  Then the following data should be generated:
+  | foo |
+  | null |
+  | "a"  |
+
+Scenario: Running a 'longerThan' request on a negative number should fail with an error
+    Given foo is longer than -5
+    Then I am presented with an error message
+    And no data is created
+
+
+Scenario: Running a 'longerThan' request on a decimal number to specify a the length of a generated string should fail with an error message
+  And foo is longer than 1.1
+  Then I am presented with an error message
+  And no data is created
+
+
+  Scenario: Running a 'longerThan' request on a decimal number to specify a the length of a generated string test2 should fail with an error message
+    And foo is longer than "Test"
+    Then I am presented with an error message
+    And no data is created
+
+  Scenario: Running a 'longerThan' request on a decimal number to specify a the length of a generated string test2 should be successful
+    And foo is longer than 2.0
+    And foo is in set:
+    | 1.0 |
+    | 2.0 |
+    | "a" |
+    | "aaa" |
+    Then the following data should be generated:
+      | foo |
+      | null |
+      | 1.0  |
+      | 2.0  |
+      | "aaa" |
+
+
+
+Scenario: Running a 'longerThan' request on a character string should be successful
+    Given foo is longer than 2
+    And foo is longer than 1
+    And foo is in set:
+      | "a" |
+      | "aa" |
+      | "aaa" |
+      | "aab" |
+    Then the following data should be generated:
+      |  foo  |
+      | null  |
+      | "aaa" |
+      | "aab" |
+
+
+Scenario: Running a 'longerThan' request on a character string should be successful
+  Given foo is longer than 1
+  And foo is anything but longer than 3
+  And foo is in set:
+   | "a" |
+   | "aa" |
+   | "aaa" |
+   | "aaaa" |
+  Then the following data should be generated:
+   | foo |
+   | null |
+   | "aa" |
+   | "aaa" |
+
+Scenario: Running a 'longerThan' request on a character string should be successful
+  Given foo is anything but longer than 5
+  And foo is anything but longer than 6
+  And foo is in set:
+    | "a" |
+    | "aa" |
+    | "aaa" |
+    | "aaaa" |
+    | "aaaaa" |
+    | "aaaaaa" |
+  Then the following data should be generated:
+    | foo |
+    | null |
+    | "a" |
+    | "aa" |
+    | "aaa" |
+    | "aaaa" |
+    | "aaaaa" |
+
+Scenario:  Running a 'longerThan' request against contradicting not longerThan constraint should only generate numeric,temporal and null
+  Given foo is longer than 3
+  And foo is anything but longer than 3
+  And foo is in set:
+  | 1 |
+  | "abc" |
+  | 2011-01-01T00:00:00.000 |
+  Then the following data should be generated:
+  | foo |
+  | null |
+  | 1    |
```

Comment:
This seems weird to me? Maybe worth raising the question if this is right - I think it's related to our laziness with type constraints

Reply:
This expected and correct behaviour given how the generator currently works. See shorterThan feature.

Typed constraints only affect values of their type. Otherwise put they only control generation of values from 'their domain' of the universal set.